scripts/sonic_sku_create.py

```diff
@@ -0,0 +1,358 @@
+#! /usr/bin/env python2
+"""
+using sonic_sku_create.py  to create a new SONiC SKU based on a <SKU-DEF>.xml file
+
+usage: sonic_sku_create.py [-h] [-v] [-f FILE] [-b BASE] [-r]
+                           [-c {new_sku_only,l2_mode_only,new_sku_l2}]
+                           [-l2_sku_name L2_SKU_NAME] [-p] [-vv]
+
+Create a new SKU
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -v, --version         show program's version number and exit
+  -f FILE, --file FILE  SKU definition file (default <cwd>/sku_def.xml)
+  -b BASE, --base BASE  SKU base definition
+  -r, --remove          Remove SKU folder
+  -c {new_sku_only,l2_mode_only,new_sku_l2}, --cmd {new_sku_only,l2_mode_only,new_sku_l2}
+                        Choose action to preform (Generate a new SKU, Configure L2 mode, Both
+  -l2_sku_name L2_SKU_NAME, --l2_sku_name L2_SKU_NAME
+                        SKU name to be used in the L2 configuration mode
+  -p, --print           Print port_config.ini without creating a new SKU
+  -vv, --verbose        Verbose output
+
+
+"""
+
+from __future__ import print_function
+
+import argparse
+import json
+import os
+import re
+import subprocess
+import swsssdk
+import sys
+import shutil
+import sonic_platform
+
+from tabulate import tabulate
+import xml.etree.ElementTree as ET
+
+
+DEFAULT_DEV_PATH = '/usr/share/sonic/device/'
+### port_config.ini header
+PORTCONFIG_HEADER = ["# name", "lanes", "speed", "alias", "index"]
+
+
+class SkuCreate(object):
+	"""
+	Process aclstat
+	"""
+
+	def __init__(self):
+		###self.metadata = {}
+		self.portconfig_dict = {}
+		self.platform_specific_dict = {"x86_64-mlnx_msn2700-r0":self.msn2700_specific}
+		self.default_lanes_per_port = []
+		self.fpp = []
+		self.fpp_split  =  {}
+		self.num_of_fpp = 0
+		self.num_of_lanes_dict = {"1" : 4, "2" : 2, "4" : 1}
+		
+		# Set up db connections
+		###self.db = swsssdk.SonicV2Connector(host="127.0.0.1")
+		###self.db.connect(self.db.CONFIG_DB)
+		
+		###self.configdb = swsssdk.ConfigDBConnector()
+		###self.configdb.connect()
+
+		###self.read_metadata()
+		try:
+			platform = subprocess.check_output("sonic-cfggen -H -v DEVICE_METADATA.localhost.platform",shell=True) #self.metadata['platform']
+			self.platform = platform.rstrip()
+		except KeyError:
+			print ("Couldn't find platform info in CONFIG_DB DEVICE_METADATA")
```

Comment:
>print [](start = 3, length = 5)

print warning or error messages to stderr, instead of stdout.

Reply:
Fixed